contracts/test/TestDelegateCallReceiver.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.18;
+
+contract TestDelegateCallReceiver {
```

Comment:
This looks like a regular receiver contract to me; why is it called delegate call receiver?

Reply:
Its named `TestDelegateCallReceiver` just to be specific for what the contract is used in the tests here.

Reply:
I would update the name to be honest, as, without context, it sounds incorrect or adjust the code contract only to accept delegate calls.